lib/widgets/bubbles.dart

```diff
@@ -5,85 +5,78 @@ import 'package:flutter/material.dart';
 import 'package:bars_frontend/main.dart';
 import 'package:bars_frontend/utils.dart';
 import 'package:bars_frontend/predictions.dart';
-import 'topBubbleBar.dart';
+import 'bubblesPrototype.dart';
 import 'dialogs.dart';
 
+/// Represents a draggable widget that includes a widget of type [FeatureBubble].
+/// [initialOffset] is the position the bubble is originally at.
+/// [bubbleWidth] is the width of the child [FeatureBubble].
+/// [labelBubbleWidth] is the width of [LabelBubble]s.
+/// [feature] is the MapEntry of the feature to be represented.
 class DragBubble extends StatefulWidget {
   final Offset initialOffset;
   final double bubbleWidth;
   final double labelBubbleWidth;
   final MyHomePageState homePageState;
   final BubblePrototypeState bubblePrototypeState;
   final MapEntry<String, dynamic> feature;
-  final Map<String, dynamic> modelConfig;
 
-  DragBubble(
-      this.initialOffset,
-      this.bubbleWidth,
-      this.labelBubbleWidth,
-      this.homePageState,
-      this.bubblePrototypeState,
-      this.modelConfig,
-      this.feature);
+  DragBubble(this.initialOffset, this.bubbleWidth, this.labelBubbleWidth,
+      this.homePageState, this.bubblePrototypeState, this.feature);
 
   @override
   State<StatefulWidget> createState() {
     return DragBubbleState(initialOffset, bubbleWidth, labelBubbleWidth,
-        homePageState, bubblePrototypeState, modelConfig, feature);
+        homePageState, bubblePrototypeState, feature);
   }
 }
 
+/// [offset] is the current position of the [DragBubble]. It is changed when the [DragBubble] is dragged.
+/// [color] is the current color of the [DragBubble]. It is changed when the input value of the associated [feature] changes.
+/// [isSmall] changes to false when the [DragBubble] is dragged the first time.
```

Comment:
```suggestion
/// [isSmall] changes to false when the [DragBubble] once it is dragged away from the top.
```

Reply:
/// [isSmall] changes to false once the [DragBubble] is dragged away from the top.